quick-start/src/main/ui/app/components/flows-new/edit-flow/mapping/ui/mapping-ui.component.ts

```diff
@@ -0,0 +1,343 @@
+  templateUrl: './mapping-ui.component.html',
+  styleUrls: ['./mapping-ui.component.scss']
+})
+export class MappingUiComponent implements OnChanges {
+
+  @Input() mapping: Mapping = new Mapping();
+  @Input() chosenEntity: Entity = null;
+  @Input() conns: object = {};
+  @Input() sampleDocSrcProps: Array<any> = [];
+  @Input() editURIVal: string = '';
+
+  @Output() updateDesc = new EventEmitter();
+  @Output() updateURI = new EventEmitter();
+  @Output() updateMap = new EventEmitter();
+  @Output() resetMap = new EventEmitter();
+
+  private uriOrig: string = '';
+  private connsOrig: object = {};
+
+  public valMaxLen: number = 17;
+
+  public filterFocus: object = {};
+  public filterText: object = {};
+
+  public editDescVal: string;
+  public editingDesc: boolean = false;
+  public editingURI: boolean = false;
+
+  /**
+   * Update the sample document based on a URI.
+   */
+  onUpdateURI() {
+    if (Object.keys(this.conns).length > 0) {
+      let result = this.dialogService.confirm(
+          'Changing your source document will remove<br/>existing property selections. Proceed?',
+          'Cancel', 'OK');
+      result.subscribe( () => {
+          this.conns = {};
+          this.editingURI = false;
+          this.updateURI.emit({
+            uri: this.editURIVal,
+            uriOrig: this.mapping.sourceURI,
+            conns: this.conns,
+            connsOrig: this.connsOrig,
+            save: true
+          });
+        },(err: any) => {
+          console.log('source change aborted');
+          this.editingURI = false;
+        },
+        () => {}
+      );
+    } else {
+      this.editingURI = false;
+      this.updateURI.emit({
+        uri: this.editURIVal,
+        uriOrig: this.mapping.sourceURI,
+        conns: this.conns,
+        connsOrig: {},
+        save: true
+      });
+    }
+  }
+
+  /**
+   * Cancel the editing of the source document URI.
+   */
+  cancelEditURI() {
+    this.editURIVal = this.mapping.sourceURI;
+    this.editingURI = false;
+  }
+
+  /**
+   * Handle "Enter" keypress for the source document URI box.
+   * @param event Event object
+   */
+  keyPressURI(event) {
+    if (event.key === 'Enter') {
+      this.onUpdateURI();
+    }
+  }
+
+  /**
+   * Handle when edit URI is not found.
+   * @param uri URI not found
+   */
+  uriNotFound(uri) {
+    let result = this.dialogService.alert(
+      'Document URI not found: ' + uri,
+      'OK'
+    );
+    result.subscribe( () => {
+        this.editURIVal = this.mapping.sourceURI;
+        // rollback to conns from previous URI
+        if (!_.isEmpty(this.connsOrig)) {
+          this.conns = this.connsOrig;
+        }
+      },
+      () => {},
+      () => {}
+    )
+  }
+
+  /**
+   * Display snackbar popup.
+   * @param msg Message text to display in popup.
+   */
+  showSnackbar(msg) {
+    this.snackbar.showSnackbar({
+      message: msg
+    });
+  }
+
+  constructor(
+    private dialogService: MdlDialogService,
```

Comment:
For dialog service, we have a ConfirmDialog in common/confirm-dialog instead of MdlDialogService